### PR TITLE
Fix error for constant declaration of invalid types

### LINKED ISSUE #30334: Fix the error for constant declarations of an invalid type
**Description:**
$title.

The spec says for a constant declaration

> The type-descriptor must specify a type that is a subtype of anydata and must not be never. 

**Describe your problem(s)**
```ballerina
const Foo FOO = new;

class Foo {
    int i = 1;
}
```

But this currently results in the following error.

```cmd
$ bal run foo.bal 

Compiling source
        foo.bal
ERROR [foo.bal:(1:7,1:10)] constant cannot be defined with type 'Foo', expected a simple basic types or a map of a simple basic type
```

With the changes to fix https://github.com/ballerina-platform/ballerina-lang/issues/30073 I think we should just say this type is invalid in a constant declaration.

### PR DESCRIPTION
**Pull request by rpjayasekara**
## Purpose

Fixes #30334 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 1]
```diff
@@ -1435,8 +1435,8 @@ error.cannot.update.constant.value=\
 error.cannot.assign.value.to.constant=\
   cannot assign a value to a constant
 
-error.cannot.define.constant.with.type=\
-  constant cannot be defined with type ''{0}'', expected a simple basic types or a map of a simple basic type
+error.invalid.const.declaration=\
+  type ''{0}'' is invalid in constant declaration
```
Comment (rdhananjaya) [inline compiler/ballerina-lang/src/main/resources/compiler.properties:1439]: Shall we have a bit more descriptive error message, something that says this type must be subtype of anydata and must not be never

Comment (rpjayasekara) [reply compiler/ballerina-lang/src/main/resources/compiler.properties:1439]: How about this? `type ''Foo'' is invalid in constant declaration, expected a simple basic type or a map of a simple basic type and must not be never`

Comment (rdhananjaya) [reply compiler/ballerina-lang/src/main/resources/compiler.properties:1439]: both maps and lists.
I prefer just saying `anydata`

Comment (rpjayasekara) [reply compiler/ballerina-lang/src/main/resources/compiler.properties:1439]: `type ''Foo'' is invalid in constant declaration, expected a subtype of anydata and must not be never` ?

File: compiler/ballerina-lang/src/main/resources/compiler.properties [hunk 2]
```diff
@@ -1444,8 +1444,8 @@ error.cannot.update.constant.value=\
 error.cannot.assign.value.to.constant=\
   cannot assign a value to a constant
 
-error.cannot.define.constant.with.type=\
-  constant cannot be defined with type ''{0}'', expected a simple basic types or a map of a simple basic type
+error.invalid.const.declaration=\
+  type ''{0}'' is invalid in constant declaration, expected a subtype of anydata and must not be never
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/resources/compiler.properties:1448]: How about
```suggestion
  cannot declare a constant with type ''{0}'', expected a subtype of ''anydata'' that is not ''never''
```

File: tests/jballerina-unit-test/src/test/resources/test-src/types/constant/simple-literal-constant-negative.bal [hunk 3]
```diff
@@ -284,3 +284,12 @@ public const int:Signed32 ONE = 1;
...
 public const string:Char A = "A";
+
+// -----------------------------------------------------------
+
+class Bar {
+    int i = 1;
+}
+
+const Bar BAR = new;
+
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/types/constant/simple-literal-constant-negative.bal:295]: Extra new line?

File: compiler/ballerina-lang/src/main/resources/compiler.properties (hunk 2 above)
Comment (rpjayasekara) [reply compiler/ballerina-lang/src/main/resources/compiler.properties:1448]: +1

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**